webapp/javascript/components/AppSelector/index.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+import { queryFromAppName, queryToAppName, Query } from '@webapp/models/query';
+import { useAppSelector, useAppDispatch } from '@webapp/redux/hooks';
+import {
+  actions,
+  selectAppNames,
+  reloadAppNames,
+  selectQueries,
+  selectAppNamesState,
+} from '@webapp/redux/reducers/continuous';
+import Button from '@webapp/ui/Button';
+import LoadingSpinner from '@webapp/ui/LoadingSpinner';
+import OutsideClickHandler from 'react-outside-click-handler';
+import { faSyncAlt } from '@fortawesome/free-solid-svg-icons/faSyncAlt';
+import SelectorModal from './SelectorModal';
+import styles from './AppSelector.module.scss';
+
+interface AppSelectorProps {
+  onSelectedName?: (name: Query) => void;
+}
+
+const TOGGLE_BTN_ID = 'toggle_button';
+
+const Loading = ({ type }: { type: 'reloading' | 'loaded' | 'failed' }) => {
+  switch (type) {
+    case 'reloading': {
+      return <LoadingSpinner />;
+    }
+
+    default: {
+      return null;
+    }
+  }
+};
+
+const AppSelector = ({ onSelectedName }: AppSelectorProps) => {
+  const dispatch = useAppDispatch();
+  const appNames = useAppSelector(selectAppNames);
+  const { query } = useAppSelector(selectQueries);
+  const appName = queryToAppName(query).mapOr('', (q) =>
+    appNames.indexOf(q) !== -1 ? q : ''
+  );
+  const appNamesState = useAppSelector(selectAppNamesState);
+
+  const [selectorOpened, toggleSelector] = useState(false);
+
+  const selectAppName = (name: string) => {
+    const appNameQuery = queryFromAppName(name);
+
+    if (onSelectedName) {
```

Comment:
This was a bit of a hack from the previous implementation IIRC, can you add a small comment explaining that we allow overwriting the behaviour when a name is selected? Could be in the props.